compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -1922,4 +1922,36 @@ object messages {
     val msg = hl"Traits cannot redefine final $method from ${"class AnyRef"}."
     val explanation = ""
   }
+
+  case class UnableToExtendSealedClass(pclazz: Symbol)(implicit ctx: Context) extends Message(UnableToExtendSealedClassID) {
+    val kind = "Syntax"
+    val msg = hl"cannot extend sealed $pclazz in different compilation unit"
+    val explanation = {
+      val errorCodeExample =
+        """//A.scala
+          |sealed class A
+          |
+          |//B.scala
+          |class B extends A""".stripMargin
+      val codeExample =
+        """//A.scala
+          |sealed class A
+          |class B extends A""".stripMargin
+
+      hl"""A ${"sealed"} keyword makes sure that all classes that extend a sealed class are
```

Comment:
TBH I'm not convinced a super long example with correct/incorrect examples is helpful here. [The top SO answer](https://stackoverflow.com/questions/11203268/what-is-a-sealed-trait) just gets to the point:

A `sealed` trait can be extended only in the same file as its declaration.